test/specs/lwcLsp.e2e.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { step } from 'mocha-steps';
+import { TextEditor } from 'wdio-vscode-service';
+import { TestSetup } from '../testSetup';
+import * as utilities from '../utilities';
+
+describe('LWC LSP', async () => {
+  let testSetup: TestSetup;
+  let projectName: string;
+
+  step('Set up the testing environment', async () => {
+    testSetup = new TestSetup('LwcLsp', false);
+    await testSetup.setUp();
+    projectName = testSetup.tempProjectName.toUpperCase();
+
+    // Create Lightning Web Component
+    await utilities.createLWC('lwc1');
+  });
+
+  step('Verify Extension is Running', async () => {
+    // Using the Command palette, run Developer: Show Running Extensions
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(
+      workbench,
+      'Developer: Show Running Extensions',
+      1
+    );
+    let buttons = await $$('a.monaco-button.monaco-text-button');
+    for (const item of buttons) {
+      const text = await item.getText();
+      if (text.includes('Install and Reload')) {
+        await item.click();
+      }
+    }
+    await utilities.pause(10);
+    buttons = await $$('a.monaco-button.monaco-text-button');
+    for (const item of buttons) {
+      const text = await item.getText();
+      if (text.includes('Reload and Enable Extensions')) {
+        await item.click();
+      }
+    }
+    await utilities.pause(5);
+    await utilities.runCommandFromCommandPrompt(
+      workbench,
+      'Developer: Reload Window',
+      10
+    );
+
+    // Verify Lightning Web Components extension is present and running
+    const extensionNameDivs = await $$('div.name');
+    let extensionWasFound = false;
+    for (const extensionNameDiv of extensionNameDivs) {
+      const text = await extensionNameDiv.getText();
+      if (text.includes('salesforce.salesforcedx-vscode-lwc')) {
+        extensionWasFound = true;
+      }
+    }
+    expect(extensionWasFound).toBe(true);
+  });
+
+  step('Go to Definition (Javascript)', async () => {
+    // Get open text editor
+    const workbench = await browser.getWorkbench();
+    const editorView = workbench.getEditorView();
+    const textEditor = (await editorView.openEditor('lwc1.js')) as TextEditor;
+    await textEditor.moveCursor(3, 40);
```

Comment:
It was really cool to put a breakpoint here and then watch the cursor move in the IDE 🪄